### PR TITLE
add highlighting for loose inter-service relations

### LINKED ISSUE #5998: Add looseElement highlighter to loose related instances

### PR DESCRIPTION
**Pull request by matborowczyk**
# Description

https://github.com/user-attachments/assets/0cb45491-8cd0-4639-a6a2-746c6a2fb845

closes #5998 

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [ ] Attached issue to pull request
- [ ] Changelog entry
- [ ] Code is clear and sufficiently documented
- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )

### CODE REVIEW

File: src/UI/Components/Diagram/Context/ComposerEditorProvider.tsx [hunk 1]
```diff
@@ -69,6 +69,7 @@ export const ComposerEditorProvider: React.FC<Props> = ({
     instance,
     environment,
     mainService,
+    !editable, //if editable is true, we don't fetch referenced_by instances as they should be displayed to keep it aligned with the regular instance form
```
Comment (matborowczyk) [inline src/UI/Components/Diagram/Context/ComposerEditorProvider.tsx:72]:
```suggestion
    !editable, //if editable is true, we don't fetch referenced_by instances as they should not be displayed to keep it aligned with the regular instance form, they are only displayed in the composer viewer
```

File: src/UI/Components/Diagram/helpers.ts [hunk 2]
```diff
@@ -929,3 +904,52 @@ export function showLinkTools(
 
   linkView.addTools(tools);
 }
+
+/**
+ * Function that remove data about niter-service relation between cells
```
Comment (matborowczyk) [inline src/UI/Components/Diagram/helpers.ts:909]:
```suggestion
 * Function that remove data about inter-service relation between cells
```

File: src/UI/Components/Diagram/helpers.ts [hunk 3]
```diff
@@ -929,3 +904,52 @@ export function showLinkTools(
...
+  disconnectingCell: ServiceEntityBlock,
+): void => {
+  const elementRelations = elementCell.getRelations();
+
+  // resolve any possible relation connections between cells
+  if (elementRelations && elementRelations.has(String(disconnectingCell.id))) {
+    elementCell.removeRelation(String(disconnectingCell.id));
+
+    //get all connected entities to inter-service relation cell (the current relation that will be removed is included here)
+    const disconnectingCellNeighbors = graph.getNeighbors(disconnectingCell);
```
Comment (LukasStordeur) [inline src/UI/Components/Diagram/helpers.ts:929]: same remark here about "disconnecting", should be disconnected

File: src/UI/Components/Diagram/helpers.ts [hunk 4]
```diff
@@ -929,3 +904,52 @@ export function showLinkTools(
...
+    const embeddedToID = disconnectingCell.get("embeddedTo");
+
+    //filter out every cell that is either embedded entity of to the disconnectingCell or is core or the disconnectingCell is embedded to it
+    const interServiceRelations = disconnectingCellNeighbors.filter(
+      (cell) =>
+        !(cell.get("embeddedTo") === disconnectingCell.id) ||
+        !(embeddedToID === cell.id),
+    );
+
+    //all left are the inter-service relation connections, if there is more than one then we should highlight the cell
```
Comment (LukasStordeur) [inline src/UI/Components/Diagram/helpers.ts:941]: the choice of name of the variable, and the comment above make it hard to follow why having more than one element should highlight the cell. I also find it odd you're using `=== 1 ` then just below 🤔 and not `< 1`

**Review by LukasStordeur — CHANGES_REQUESTED** [review]

File: src/Data/Managers/V2/GETTERS/GetInstanceWithRelations/useGetInstanceWithRelations.ts [hunk 5]
```diff
@@ -29,13 +30,15 @@ interface GetInstanceWithRelationsHook {
...
- * @param {ServiceModel} serviceModel - The service Model of the instance (optional as it can be undefined at the init of the component that use the hook)
+ * @param {ServiceModel} [serviceModel] - The service Model of the instance (optional as it can be undefined at the init of the component that use the hook)
+ * @param {boolean} [referenced_by] - A flag indicating if we should fetch instances that reference our main instance - defaults to false.
  * @returns  {GetInstanceWithRelationsHook} An object containing a custom hook to fetch the instance with its related instances.
  */
 export const useGetInstanceWithRelations = (
   instanceId: string,
   environment: string,
   serviceModel?: ServiceModel,
+  referenced_by: boolean = false,
```
[the remark this answers is not in the document]
Comment (matborowczyk) [reply src/Data/Managers/V2/GETTERS/GetInstanceWithRelations/useGetInstanceWithRelations.ts:41]: 1. Parameters can't have question mark and initializer. So it's being treated as optional, and the order here is fine.
2. I'll rename it but I thought it's clear as that property is going directly to the api endpoint

File: src/UI/Components/Diagram/helpers.ts (hunk 4 above)
Comment (matborowczyk) [reply src/UI/Components/Diagram/helpers.ts:941]: improved comment, should be, if there is only one cell - as we are including the one that we are trying to disconnect. so 0 should not be possible here, if there is more than 1, which can theoretically happen, then we don't highlight that cell as it's already connected

File: src/Data/Managers/V2/GETTERS/GetInstanceWithRelations/useGetInstanceWithRelations.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (LukasStordeur) [reply src/Data/Managers/V2/GETTERS/GetInstanceWithRelations/useGetInstanceWithRelations.ts]: In JavaScript (and TypeScript), arguments with default values are often considered "required" but with an initial default, meaning they provide a sensible value even if not explicitly passed. 

We tend to think of "optional" as "not required at all," but in reality, any argument with a default has a specific intention and should be thought of as a parameter with a baseline, and thus precede the optional arguments in the order of declaration.

File: src/Data/Managers/V2/GETTERS/GetInstanceWithRelations/useGetInstanceWithRelations.ts [hunk 6]
```diff
@@ -29,15 +29,15 @@ interface GetInstanceWithRelationsHook {
  * React Query hook to fetch an instance with its related instances from the API. The related instances are all instances connected with given instance by inter-service relation, both, as a parent and as a child.
  * @param {string} id - The ID of the instance to fetch.
  * @param {string} environment - The environment in which we are looking for instances.
+ *  * @param {boolean} [includesReferencedBy] - A flag indicating if we should fetch instances that reference our main instance - defaults to false.
```
Comment (LukasStordeur) [inline src/Data/Managers/V2/GETTERS/GetInstanceWithRelations/useGetInstanceWithRelations.ts:32]: Update docstrings too =) you still flagged it as optional, and have double asterix

**Review by LukasStordeur — CHANGES_REQUESTED** [review]

**Review by LukasStordeur — APPROVED** [review]
